kairon/shared/data/processor.py

```
@@ -254,6 +254,211 @@
+    @staticmethod
+    def is_slot_in_pyscript_action_config(config: dict, slot: Text):
+        import re
+
+        source_code = config.get('source_code', '')
+        slot_pattern = re.compile(r'slot\[\s*[\'\"]{}[\'\"]\s*\]'.format(re.escape(slot)))
+        slots_dict_pattern = re.compile(r'slots\s*=\s*\{{.*[\'\"]{}[\'\"]\s*:'.format(re.escape(slot)))
+
+        if slot_pattern.search(source_code) or slots_dict_pattern.search(source_code):
+            return True
+
+        return False
```

Ruff

Return the condition directly

Inline condition

(SIM103)